rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,477 @@
+            raise FTXLoginError('Authentication failed.')
+
+        if response.status_code != 200:
+            raise RemoteError(
+                f'FTX query {full_url} responded with error status code: '
+                f'{response.status_code} and text: {response.text}',
+            )
+
+        try:
+            json_ret = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            raise RemoteError(f'FTX returned invalid JSON response: {response.text}') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'FTX json response does not contain data: {response.text}')
+
+        return json_ret['result']
+
+    def _api_query(
+            self,
+            endpoint: str,
+            start_time: int = None,
+            end_time: int = None,
+            limit: int = 100,
+            paginate: bool = True,
+    ) -> Any:
+        if not paginate:
+            return self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=end_time,
+            )
+
+        # If there is pagination
+        new_end_time = end_time
+        ids = set()
+        final_data = []
+        while True:
+            step = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=new_end_time,
+            )
+
+            if not isinstance(step, list):
+                self.msg_aggregator.add_error('Trying pagination in an invalid endpoint.')
+                break
+
+            # remove possible duplicates
+            deduped = [r for r in step if 'id' in r.keys() and r['id'] not in ids]
+            ids |= {r['id'] for r in deduped}
+            final_data.extend(step)
+
+            if len(step) == 0:
+                break
+
+            # Avoid error if there is an error in any date
+            times = []
+            for t in step:
+                try:
+                    times.append(deserialize_timestamp_from_date(t['time'], 'iso8601', 'ftx'))
+                except DeserializationError:
+                    continue
+
+            if len(times):
+                new_end_time = min(times)
+            else:
+                self.msg_aggregator.add_error(
+                    'Error processing ftx trade history. Fill query step '
+                    'returned invalid entries.',
+                )
+                break
+
+            if len(step) < limit:
+                break
+
+        return final_data
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        try:
+            resp = self._api_query('wallet/all_balances', paginate=False)
+        except RemoteError as e:
+            msg = (
+                'FTX API request failed. Could not reach ftx due '
+                'to {}'.format(e)
+            )
```

Comment:
use f strings

Reply:
This one is in the other exchanges too. Maybe I should take note of this comments and save for a possible `fixing the style PR`

Reply:
style changes like that can just be done when touching the code in question. There is really no reason to waste resources/time in specifically trying to find and fix.